server/src/storage/data_transfer.rs

```diff
@@ -0,0 +1,399 @@
+    }
+
+    #[test]
+    fn test_add_compressed_file_into_new_key() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (_target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+        let parquet_path = create_compressed_file(temp_dir.path(), "test");
+
+        assert!(data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).is_ok());
+        assert_eq!(data_transfer.compressed_files.get(&KEY).unwrap(), &COMPRESSED_FILE_SIZE);
+    }
+
+    #[test]
+    fn test_add_compressed_file_into_existing_key() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (_target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+        let parquet_path = create_compressed_file(temp_dir.path(), "test");
+
+        data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).unwrap();
+        data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).unwrap();
+
+        assert_eq!(data_transfer.compressed_files.get(&KEY).unwrap(), &(COMPRESSED_FILE_SIZE * 2));
+    }
+
+    #[test]
+    fn test_add_non_existent_file() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let path = temp_dir.path().join(format!("{}/compressed", KEY));
+        fs::create_dir_all(path.clone()).unwrap();
+
+        let (_target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+
+        let parquet_path = path.join("test_parquet.parquet");
+        assert!(data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).is_err())
+    }
+
+    #[test]
+    fn test_transfer_single_file() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+        let parquet_path = create_compressed_file(temp_dir.path(), "test");
+
+        data_transfer.add_compressed_file(&KEY, parquet_path.as_path()).unwrap();
+        data_transfer.transfer_data(&KEY).unwrap();
+
+        // The transferred file should have a time range file name that matches the compressed data.
+        assert!(target_dir.path().join(format!("{}/compressed/0-3.parquet", KEY)).exists());
+        assert_eq!(*data_transfer.compressed_files.get(&KEY).unwrap(), 0 as usize);
+    }
+
+    #[test]
+    fn test_transfer_multiple_files() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let (target_dir, mut data_transfer) = create_data_transfer_component(temp_dir.path());
+        let path_1 = create_compressed_file(temp_dir.path(), "test_1");
+        let path_2 = create_compressed_file(temp_dir.path(), "test_2");
+
+        data_transfer.add_compressed_file(&KEY, path_1.as_path()).unwrap();
+        data_transfer.add_compressed_file(&KEY, path_2.as_path()).unwrap();
+        data_transfer.transfer_data(&KEY).unwrap();
+
+        assert!(!path_1.exists());
+        assert!(!path_2.exists());
+
```

Comment:
Maybe add comment from Line 312 in `test_transfer_single_file()` to make it clear what is checked.

Reply:
Fixed.